docs/advanced-topics/snapshot-releases.md

```diff
@@ -0,0 +1,131 @@
+# Snapshot releases
+
+In addition to official JanusGraph releases, JanusGraph publishes snapshot 
+releases for each commit. The snapshot releases allow users to use latest 
+JanusGraph features without relying on official JanusGraph releases.  
+- Official JanusGraph releases are better tested and usually comes with finilized 
+changes which signals that the used features and most likely to be stable for long term.
+- Snapshot releases are not manually verified but instead verified by main CI tests as
+well as scheduled full CI tests (once per week). New features in snapshot releases are not 
+guaranteed to be compatible between commits. Thus, you may expect more breaking changes between 
+snapshot releases and official releases.
+
+## Maven repository artifacts
+
+Official release are deployed to Sonatype OSS and are available in Sonatype Maven Central Repository. 
+No authentication is needed to use such releases.  
+
+Snapshot releases are deployed to GitHub Packages and are available in 
+[JanusGraph GitHub Packages registry](https://github.com/orgs/janusgraph/packages?repo_name=janusgraph). 
+Thus, such release are available only via GitHub authentication.  
+
+To use snapshot releases in your Maven or Gradle build automation tool the next steps are required:
+1. Generate GitHub token with read access to packages and public repository access.
+2. Add JanusGraph GitHub Packages repository with higher priority to Maven Central repository. 
+
+#### GitHub token generation
+
+See detailed instruction on how to [create a personal access token in GitHub](https://docs.github.com/en/enterprise-server@3.4/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token).
+
+Currently, to install public packages GitHub requires authentication. The minimum scope token needs to have to  
+be able to install public packages is: `packages:read` and `public_repo`. When you create the token, notice that by default 
+expiration time may be set which will prevent usage of this token after the expiration time. You may specify `No expiration` 
+in case you don't want your token to be expired in the future.
+
+To create a token you can click on your profile -> Settings -> Developer settings -> Personal access tokens -> 
+Tokens (classic) -> Generate new token (classic).
+
+![](github_read_packages_token.png)
+
+After the token is generated it can be used to install JanusGraph public snapshot packages.
+
+#### Placing JanusGraph GitHub Packages repository
+
+Please, see detailed instruction on how to [define GitHub Package registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-apache-maven-registry#authenticating-to-github-packages).
+
+JanusGraph GitHub Packages repository after Maven Central Repository to have a higher priority. Otherwise the official 
```

Comment:
```suggestion
[Define/Place] JanusGraph GitHub Packages repository after Maven Central Repository to have a higher priority. Otherwise the official 
```